src/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.hpp

```diff
@@ -94,56 +128,39 @@ class HalfSpaceMirror {
-                  size_t no_intervals, double absolute_tolerance) noexcept;
-
-  HalfSpaceMirror(double beam_width, double applied_force, double bulk_modulus,
-                  double shear_modulus, size_t no_intervals,
-                  double absolute_tolerance) noexcept;
+                  size_t no_intervals = 150, double absolute_tolerance = 1e-12,
+                  double relative_tolerance = 1e-10) noexcept;
 
   const constitutive_relation_type& constitutive_relation() const noexcept {
     return constitutive_relation_;
   }
 
   // @{
   /// Retrieve variable at coordinates `x`
-  auto variables(const tnsr::I<DataVector, dim>& x,
-                 tmpl::list<Tags::Displacement<dim>> /*meta*/) const noexcept
-      -> tuples::TaggedTuple<Tags::Displacement<dim>>;
-
-  auto variables(const tnsr::I<DataVector, dim>& x,
-                 tmpl::list<Tags::Strain<dim>> /*meta*/) const noexcept
-      -> tuples::TaggedTuple<Tags::Strain<dim>>;
+  auto variables(const tnsr::I<DataVector, 3>& x,
+                 tmpl::list<Tags::Displacement<3>> /*meta*/) const noexcept
+      -> tuples::TaggedTuple<Tags::Displacement<3>>;
 
-  auto variables(const tnsr::I<DataVector, dim>& x,
-                 tmpl::list<Tags::Stress<dim>> /*meta*/) const noexcept
-      -> tuples::TaggedTuple<Tags::Stress<dim>>;
+  auto variables(const tnsr::I<DataVector, 3>& x,
+                 tmpl::list<Tags::Strain<3>> /*meta*/) const noexcept
+      -> tuples::TaggedTuple<Tags::Strain<3>>;
 
   static auto variables(
-      const tnsr::I<DataVector, dim>& x,
-      tmpl::list<
-          ::Tags::FixedSource<Tags::Displacement<dim>>> /*meta*/) noexcept
-      -> tuples::TaggedTuple<::Tags::FixedSource<Tags::Displacement<dim>>>;
+      const tnsr::I<DataVector, 3>& x,
+      tmpl::list<::Tags::FixedSource<Tags::Displacement<3>>> /*meta*/) noexcept
+      -> tuples::TaggedTuple<::Tags::FixedSource<Tags::Displacement<3>>>;
   // @}
 
-  /// Initial guess for variables
-  static auto variables(
-      const tnsr::I<DataVector, dim>& x,
-      tmpl::list<::Tags::Initial<Tags::Displacement<dim>>> /*meta*/) noexcept
-      -> tuples::TaggedTuple<::Tags::Initial<Tags::Displacement<dim>>>;
-
-  static auto variables(
-      const tnsr::I<DataVector, dim>& x,
-      tmpl::list<::Tags::Initial<Tags::Strain<dim>>> /*meta*/) noexcept
-      -> tuples::TaggedTuple<::Tags::Initial<Tags::Strain<dim>>>;
-
   /// Retrieve a collection of variables at coordinates `x`
   template <typename... Tags>
-  tuples::TaggedTuple<Tags...> variables(const tnsr::I<DataVector, dim>& x,
+  tuples::TaggedTuple<Tags...> variables(const tnsr::I<DataVector, 3>& x,
                                          tmpl::list<Tags...> /*meta*/) const
       noexcept {
-    static_assert(sizeof...(Tags) > 1, "An unsupported Tag was requested.");
+    static_assert(sizeof...(Tags) > 1,
+                  "The generic template will recurse infinitely if only one "
+                  "tag is being retrieved.");
```

Comment:
please change the message back to `An unsupported Tag was requested.`, that's way more useful